File: src/1122.py
from collections import Counter

class Solution:
    def relativeSortArray(self, arr1: list[int], arr2: list[int]) -> list[int]:
        arr = []
        nums = Counter(arr1)
        for n in arr2:
            arr += [n for i in range(nums[n])]
            del nums[n]
        for n in sorted(nums):
            arr += [n for i in range(nums[n])]
        return arr

File: 1122.cpp
#include <vector>
#include <map>

using namespace std;

class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        map<int, int> nums;
        vector<int> arr;
        for(int n : arr1) {
            if(nums.count(n)) {
                nums[n] += 1;
            }else{
                nums[n] = 1;
            }
        }
        for(int n : arr2) {
            for(int i = 0; i < nums[n]; ++i) {
                arr.push_back(n);
            }
            nums.erase(n);
        }
        for(auto [k, v] : nums) {
            for(int i = 0; i < v; ++i) {
                arr.push_back(k);
            }
        }
        return arr;
    }
};
